### PR TITLE
Update CircleCI script to fix wrong iOS sim usage

### PR DESCRIPTION
**Pull request by ealeksandrov**
## Description

I've noticed that our UI tests jobs (for "iPhone 11" and "iPad Air 4") both output ([example here](https://app.circleci.com/pipelines/github/woocommerce/woocommerce-ios/10897/workflows/e25833d5-4260-4506-98b0-f86ebc7bd617/jobs/19663/parallel-runs/0/steps/0-108)):
```
Found simulator "iPhone 8 (13.5)"
```
It's not even iOS 14! 

After a small investigation I found similar issue with a fix already applied in WordPress-iOS:  https://github.com/wordpress-mobile/WordPress-iOS/pull/15261. Check for additional details and discussion there, since this PR is almost exact duplicate. Thanks @guarani!

## Changes

- Removed `ios/boot-simulator` and `ios/wait-for-simulator` steps
- Instead added `device` and `deployment_target_version` arguments to fastlane actions `run_tests` and `multi_scan`
- Cleaned up parameters everywhere to keep Xcode version, devices and iOS versions list in single place on top

## Test

1. Open "Build Tests" job
  a. Check "Build for Testing" step -> "Summary for scan" section -> ensure that the `device` parameter is "iPhone 11"
  b. Ensure the job completes successfully
2. Open "Unit Tests" job
  a. Expect to no longer see "Boot simulator" or "Wait for simulator" steps
  b. Check "Run Unit Tests" step -> search for "SimDevice" -> ensure value is `iPhone 11 ... iOS 14.0`
  c. Ensure job completes successfully
3. Open "UI Tests" jobs for iPhone and iPad
  a. Expect to no longer see "Boot simulator" or "Wait for simulator" steps
  b. Check "Run UI Tests" step -> search for "SimDevice" -> ensure value is `iPhone 11 ... iOS 14.0` and `iPad Air (4th generation) ... iOS 14.0` according to job device
  c. Ensure job completes successfully
4. Open "Installable Build" job
  a. Ensure the job completes successfully

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AliSoftware** [comment]
Overall the code LGTM, thanks for tackling this! 👍 👌 

I'll wait for you to (1) undraft this PR when ready (2) list the test steps (probably copy/paste from the WP PR) in the PR description, before submitting an actual GH review 🙂 (including maybe create a dummy commit that would make the test fail on purpose to validate the "test tailed" slack messages work too?)

**Comment by ealeksandrov** [comment]
Thanks for hints @AliSoftware! PR is ready for review now.

I've waited in draft to see output of all jobs. Probably I should have assigned reviewers only after it's green and ready 😅
Slack triggering workflow is here: https://app.circleci.com/pipelines/github/woocommerce/woocommerce-ios/10899/workflows/5423195a-a856-486d-99e3-ac1dfbb8bfcb, message is here: p1616764775224400-slack-C6H8C3G23

One small difference is that WordPress-iOS test runs have `Booted` state of sim, while Woo-iOS has `Shutdown` state. No impact on tests, but maybe there are some other differences missed between test setups.

File: .circleci/config.yml
Comment (mokagio) [inline .circleci/config.yml]: To test this PR, I changed the iPhone version from iPhone 11 to iPhone 12. When I went to unblock the UI tests, I was surprised for a second they mentioned "iPhone 11".

I'd suggest removing the device name from the job name.


```suggestion
          name: UI Tests iPad
```

File: .circleci/config.yml
Comment (mokagio) [inline .circleci/config.yml]: As mentioned above

```suggestion
          name: Optional UI Tests iPhone
```

**Review by mokagio — APPROVED** [review]
I tested it with [a commit that changed the iPhone device to "iPhone 12"](https://github.com/woocommerce/woocommerce-ios/commit/affbcf5f5f6b768c662da2fd4b08a9300a0a806c). I didn't take the time to change the iOS version.

✅ I can see "Build Tests" uses the device value I [specified](https://github.com/woocommerce/woocommerce-ios/commit/affbcf5f5f6b768c662da2fd4b08a9300a0a806c) in `iphone_test_device`

image

✅ So did the "Unit Tests" step

image

✅ The UI tests attempted to use the given devices and OS (more below on the attempted bit)

✅ The Installable Build worked, but that had no dependency on the changes from this PR 👍 

Approving this to avoid delays. I'd like the suggestion to remove the device name from the UI tests job name to be considered before merging, though.

---

The UI tests for the iPhone 12 failed to find the Simulator version

image

I _think_ that's because the GM build didn't have the image given that it was released before the new devices were announced? I did not take the time to verify this, though.

Given Apple has released Xcode 12.4 a while back, I think we should open a followup PR that upgrades CI to use 12.4.

**Comment by ealeksandrov** [comment]
@mokagio thanks for review and testing! I've removed hardcoded devices names in bd49fb7.

I agree about separate Xcode version bump (and maybe iOS too).

**Pull request merged**
